cmd/oras/root/file.go

```diff
@@ -62,3 +64,15 @@ func loadFiles(ctx context.Context, store *file.Store, annotations map[string]ma
 	}
 	return files, nil
 }
+
+func addFile(ctx context.Context, store *file.Store, name string, mediaType string, filename string) (ocispec.Descriptor, error) {
+	file, err := store.Add(ctx, name, mediaType, filename)
+	if err != nil {
+		var pathErr *fs.PathError
+		if errors.As(err, &pathErr) {
+			err = fmt.Errorf("%s: no such file or directory", pathErr.Path)
```

Comment:
This is incorrect. `*fs.PathError` can contain other types error. For example, insufficient permissions.

Repro steps:
```console
$ oras version
Version:        1.1.0+unreleased
Go version:     go1.21.0
Git commit:     2d8dcd1a7f8231a22ac00689a64aba94f0da6d65
Git tree state: clean
$ echo hello world > hello.txt
$ chmod 000 hello.txt
$ oras push localhost:5000/hello:test hello.txt
Error: failed to generate descriptor from /home/shizh/lab/test/hello.txt: open /home/shizh/lab/test/hello.txt: permission denied
```

Reply:
In this case I think printing the operation code is also necessary, otherwise user won't be able to know what permission is denied.